py-coolprop: submission of CoolProp Python module

jsalort opened the pull request:
#### Description

Submission of new port for the CoolProp Python wrapper.

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 12.3.1 21E258 arm64
Xcode 13.3.1 13E500a

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

The test suite is broken.

mascguy reviewed (approved):
LGTM. Anyone else?

reneeotten commented on python/py-coolprop/Portfile:
```diff
@@ -0,0 +1,60 @@
+PortGroup           python 1.0
+PortGroup           github 1.0
+
+github.setup        CoolProp CoolProp 6f4dfa82c1232f4827a81040284262a7099ed3fe
```
this should take a stable released version, not a git hash

reneeotten replied:
also, please take the sources from PyPI instead, which avoids fetching submodules that we like to avoid whenever possible. I haven't looked in detail what it's trying to fetch/build from submodules, but at least some of these packages are part of MacPorts already and these should preferably be used instead.


In other words: remove the `github` PG and do instead:
```
name                py-coolprop
python.rootname     CoolProp
version             6.4.1
```

of course that will require updating the checksums and let's see how that goes.

jsalort replied:
I just tried fetching 6.4.1 from PyPI as you suggest, but I get the same result I had when I originally tried to install from pip. There are several compile errors. It is my understanding that those have been fixed in the Github repository but haven't been pushed to PyPI. This is the reason why I attempted to compile from source using MacPorts in the first place. I chose this git hash because it was the most recent when I initially compiled manually from source, and I fully tested it.

It seems the latest stable tag is v6.4.1 which is presumably the same as the one from PyPI.

reneeotten replied:
then it seems time for upstream to release a new version with these fixes ;) 

We still don't want a semi-random git hash; ideally one would pick version 6.4.1 from PyPI and add the fixes as a patch to get it to build if the changes are small.

reneeotten replied:
I'll try and take a look one of these days...

mascguy replied:
> I'll try and take a look one of these days...

Renee, where do things stand with this?

jsalort replied:
FYI, I have reached out to upstream and asked them to release a new version.
See: https://github.com/CoolProp/CoolProp/issues/2178

reneeotten commented on python/py-coolprop/Portfile:
```diff
@@ -0,0 +1,60 @@
+PortGroup           github 1.0
+
+github.setup        CoolProp CoolProp 6f4dfa82c1232f4827a81040284262a7099ed3fe
+revision            1
```
this is a new submission, so the `revision` field should be zero

reneeotten commented on python/py-coolprop/Portfile:
```diff
@@ -0,0 +1,60 @@
+categories-append   science
+license             MIT
+maintainers         openmaintainer @jsalort
+platforms           darwin
```
this is the default value for the `platforms` field since MacPorts v2.7 so the line can be removed

reneeotten commented on python/py-coolprop/Portfile:
```diff
@@ -0,0 +1,60 @@
+        port:py${python.version}-setuptools \
+        port:py${python.version}-cython
+    
+    depends_test-append \
```
are these really only test dependencies and none of them are runtime dependencies? 

If they are indeed only needed for tests then you should just remove them since you do not define a test-phase (also then delete the commented out test-phase stuff)

jsalort replied:
I am not sure what is the policy in this case. They are dependencies useful at runtime naturally. But they are not hard dependencies. They are needed only if one uses the GUI functions of CoolProp (which I don't personnaly). It happens that the test suite also tests the GUI, therefore they are required for the test suite to pass.

However, there are a number of issues with the test suites, as well as the GUI. I am under the impression that the GUI and the test suite are not well maintained, and I am not convinced of their usefulness in the present state: they have errors due to outdated Python 2.7 code. I started to have a look (hence the patches) and I thought of submitting them upstream eventually, but I am not sure if it is really worth the work.

I thought I would let them commented out, so that it is easier if someone eventually decides to try again fixing the test suite. But if it is not the policy to leave half-working things commented out, then I can certainly remove them.

jsalort commented:
Upstream has now created a new release. I have updated the Portfile to use the release tarball version 6.4.3.
I also replaced the external libraries previously obtained via git submodules (which no longer existed in the release tarball) with port dependencies. This required to work on several dependencies. So I created independant PR for those requirements:

- Fix of rapidjson-devel port: https://github.com/macports/macports-ports/pull/17038
- Submission of libfmt4 port: https://github.com/macports/macports-ports/pull/17039
- Submission of coolprop-if97 port: https://github.com/macports/macports-ports/pull/17040
- Submission of coolprop-refprop-headers port: https://github.com/macports/macports-ports/pull/17041

reneeotten commented on python/py-coolprop/Portfile:
```diff
@@ -0,0 +1,57 @@
+                    size    8021549
+
+python.versions     38 39 310 311
+python.pep517       no
```
this is the default and can this be removed

reneeotten reviewed (changes requested):
one minor changes and can you then rebase with master and force-push to this PR. In that way the CI can attempt to build the port since the PRs with dependents are now merged.

reneeotten commented:
thanks @jsalort!
